framework/include/base/Assembly.h

```diff
@@ -1804,12 +1812,21 @@ class Assembly
   /// qrules() function.
   std::unordered_map<SubdomainID, std::vector<QRules>> _qrules;
 
+  /// This is an abstraction over the internal qrules function.  This is
+  /// necessary for faces because (nodes of) faces can exists in more than one
+  /// subdomain.  When this is the case, we need to use the quadrature rule from
+  /// the subdomain that has the highest specified quadrature order.  So when
+  /// you need to access a face quadrature rule, you should retrieve it via this
+  /// function.
+  QBase * qruleFace(const Elem * face);
```

Comment:
We typically do not have Elem for a face, but instead refer to a face by something like `(const Elem * elem, unsigned int side)`.

Reply:
This was in order to use the existing node block ids function on moosemesh.  We may want to come up with a better alternative to that function though.